hy/importer.py

```diff
@@ -235,3 +251,47 @@ def find_module(self, fullname, path=None):
 
 sys.meta_path.insert(0, MetaImporter())
 sys.path.insert(0, "")
+
+
+def is_package(module_name):
+    mpath = os.path.join(*module_name.split("."))
+    for path in map(os.path.abspath, sys.path):
+        if os.path.exists(os.path.join(path, mpath, "__init__.hy")):
+            return True
+    return False
+
+
+def get_bytecode_path(source_path):
+    if PY34:
+        import importlib.util
+        return importlib.util.cache_from_source(source_path)
+    elif PY3 or DEBIAN_PYPY_2:
+        # The Debian people, in their infinite wisdom, have partly backported
+        # PEP 3147 to PyPy 2.7.
+        #
+        # Whether we're on Debian PyPy 2.7 or Python 3.3, we don't have
+        # cache_from_source, so we'll fake it.
+        d, f = os.path.split(change_hy_ext_to_pyc(source_path))
+        return os.path.join(d, "__pycache__", f[:-len(".pyc")] + "." +
+                            CACHE_TAG + ".pyc")
+    else:
+        return change_hy_ext_to_pyc(source_path)
+
+
+def change_hy_ext_to_pyc(source_path):
+    """If source_path has a file extension, replace it with ".pyc".
+    Otherwise, just append ".pyc"."""
+    d, f = os.path.split(source_path)
+    return os.path.join(d, re.sub(r"(?:\.[^.]+)?\Z", ".pyc", f))
+
+
+DEBIAN_PYPY_2 = (not PY3 and platform.python_implementation() == 'PyPy'
+                 and 'dfsg' in platform.python_build()[0])
+if PY3:
+    CACHE_TAG = sys.implementation.cache_tag
+elif DEBIAN_PYPY_2:
+    # Trust me, if I knew a less stupid way to guess CACHE_TAG here, I'd do it.
```

Comment:
Won't this outright break if you run Python with `-B` or `PYTHONDONTWRITEBYTECODE=yes`?